<?xml version="1.0"?>
<launch>
<!--rosbag#####################################################################################################-->
    <param name="/use_sim_time" value="true" />
    <node pkg="rosbag" type="play" name="rosbag" args="$(find flvis)/bag/EUROC/MH_05_difficult.bag --clock" />
    <!--    V1_01_easy      -->
    <!--    V1_02_medium    -->
    <!--    V1_03_difficult -->
    <!--    V2_01_easy      -->
    <!--    V2_02_medium    -->
    <!--    V2_03_difficult -->
    <!--    MH_01_easy      -->
    <!--    MH_02_easy      -->
    <!--    MH_03_medium    -->
    <!--    MH_04_difficult -->
    <!--    MH_05_difficult -->


<!--FLVIS######################################################################################################-->
    <arg name="node_start_delay" default="0.0" />
    <param name="/yamlconfigfile" type="string" value="$(find flvis)/launch/EuRoC_MAV/euroc.yaml" />
    <param name="/voc"            type="string" value="$(find flvis)/voc/voc_orb.dbow3" />
    <!--In lite version, the visualization will be simplified -->

    <!-- Manager -->
    <node pkg="nodelet" type="nodelet"
          name="flvis_nodelet_manager" args="manager" output="screen"
          launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
          <param name="num_worker_threads" value="4" />
    </node>

    <!-- TrackingNode -->
    <node pkg="nodelet" type="nodelet" args="load flvis/TrackingNodeletClass flvis_nodelet_manager"
          name="TrackingNodeletClass_loader" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
        <!-- remap EuRoC_MAV input -->
        <remap from="/vo/input_image_0" to="/cam0/image_raw" />
        <remap from="/vo/input_image_1" to="/cam1/image_raw" />
        <remap from="/imu" to="/imu0" />
        <!-- remap output -->
        <!--<remap from="imu_pose" to="/mavros/vision_pose/pose"/>-->
        <!--<remap from="imu_odom" to="/flvis/imu_odom"/>-->
        <!--<remap from="imu_path" to="/flvis/imu_path"/>-->
    </node>

    <!-- LocalMapNode -->
    <node pkg="nodelet" type="nodelet" args="load flvis/LocalMapNodeletClass flvis_nodelet_manager"
          name="LocalMapNodeletClass_loader" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
    </node>

    <!-- LoopClosingNode -->
<!--    <node pkg="nodelet" type="nodelet" args="load flvis/LoopClosingNodeletClass flvis_nodelet_manager"
        name="LoopClosingNodeletClass_loader" output="screen"
        launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
    </node>-->


<!--output######################################################################################################-->
    <node pkg="flvis" type="vo_repub_rec" name="euroc2file" output="screen">
        <!--Sub Support Type:-->
        <!--"TransformStamped"-->
        <!--"PointStamped"-->
        <!--"PoseStamped"-->
        <!--<param name="sub_type" type="string" value="PoseStamped" />
        <param name="sub_topic" type="string" value="/imu_pose" />-->
        <!--For EuRoC_MAV V* Dataset-->
        <!--<param name="sub_type" type="string" value="TransformStamped" />
        <param name="sub_topic" type="string" value="/vicon/firefly_sbx/firefly_sbx" />-->
        <!--For EuRoC_MAV MH* Dataset-->
        <param name="sub_type" type="string" value="PointStamped" />
        <param name="sub_topic" type="string" value="/leica/position" />

        <!--Support Type:-->
        <!--"0" disable the republish function -->
        <!--"Path"-->
        <!--"PoseStamped"-->
        <param name="repub_type" type="string" value="0" />
        <param name="repub_topic" type="string" value="/republish_path" />
        <!--output_file_path = "0" disable the file output function-->
        <param name="output_file_path" type="string" value="$(find flvis)/results/euroc_gt.txt" />
    </node>

    <node pkg="flvis" type="vo_repub_rec" name="vo2file" output="screen">
        <!--Sub Support Type:-->
        <!--"TransformStamped"-->
        <!--"PointStamped"-->
        <!--"PoseStamped"-->
        <param name="sub_type" type="string" value="PoseStamped" />
        <param name="sub_topic" type="string" value="/imu_pose" />
        <!--For EuRoC_MAV V* Dataset-->
        <!--<param name="sub_type" type="string" value="TransformStamped" />
        <param name="sub_topic" type="string" value="/vicon/firefly_sbx/firefly_sbx" />-->
        <!--For EuRoC_MAV MH* Dataset-->
        <!--<param name="sub_type" type="string" value="PointStamped" />
        <param name="sub_topic" type="string" value="/leica/position" />-->

        <!--Support Type:-->
        <!--"0" disable the republish function -->
        <!--"Path"-->
        <!--"PoseStamped"-->
        <param name="repub_type" type="string" value="0" />
        <param name="repub_topic" type="string" value="/republish_path" />
        <!--output_file_path = "0" disable the file output function-->
        <param name="output_file_path" type="string" value="$(find flvis)/results/est.txt" />
    </node>


</launch>
